include/nccl_ofi_rdma.h

```diff
@@ -345,12 +345,18 @@ typedef struct nccl_net_ofi_rdma_send_comm {
 	/* Comm ID provided by remote endpoint */
 	uint64_t remote_comm_id;
 
+	/* Request tracking connect message */
+	nccl_net_ofi_rdma_req_t *send_conn_req;
+
+	/* Indicates if connect message was delivered (and req freed) */
+	bool connect_msg_delivered;
+
 	/* Request to receive connect response message to finalize
 	 * connection establishment */
 	nccl_net_ofi_rdma_req_t *conn_resp_req;
 
-	/* Indicates if connection establishment is completed */
-	bool connected;
+	/* Indicates if connection response received and connection finalized */
+	bool connect_resp_finalized;
```

Comment:
What does "connect is finalized" mean?  That's not super clear.  What is the meaning of this particular flag?  it sounds like it has two meanings?

Reply:
Yeah, this flag is really tracking when the send-comm has all its rail information initialized, which happens in function `finish_connect` after receiving the connect response message. So I was trying to express both things in the name, but it isn't a very good name.

I'm open to better names. I thought about `connect_resp_received`, but that doesn't express the "rail information initialized" part of the flag.

Reply:
Why try to express two states in one go?

Reply:
We could split it into, say, `connect_resp_received` and `connect_finalized_rail_info` (not sure I really like the latter name), but I guess my objection there is that the first flag isn't really useful in itself. As soon as we receive the connect response info, we finish initializing the rail info, so these two flags will pretty much always have the same value. Thus it makes more sense to have these be one flag/state.

But no strong feeling here; if you want to have two separate flags I can do that.

Reply:
then it's one thing the flag is telling you, and this is just a bad name.  But I'm still not entirely sure I understand what you're trying to say is done with this flag?

Reply:
When we receive the connection response message (https://github.com/aws/aws-ofi-nccl/blob/02dc354bab2c145f8cf4c91c6fb12ffb9f7df16d/src/nccl_ofi_rdma.c#L1410), we immediately call `finish_connect`, which initializes the rail info and sets this flag (called `connect_resp_finalized` in this PR) to true.

So to me, this flag is indicating that the send comm is fully initialized and ready to use, and that is how we use this flag in the code.

Reply:
Renamed connect_finalized.